packages/lesswrong/components/comments/CommentsViews.tsx

```diff
@@ -10,19 +10,37 @@ import type { Option } from '../common/InlineSelect';
 import { getCommentViewOptions } from '../../lib/commentViewOptions';
 import { useNavigate } from '../../lib/reactRouterWrapper';
 
-const CommentsViews = ({post}: {post?: PostsDetails}) => {
+const CommentsViews = ({post, setRestoreScrollPos}: {post?: PostsDetails, setRestoreScrollPos?: Function}) => {
   const currentUser = useCurrentUser();
   const navigate = useNavigate();
   const location = useLocation();
   const { query } = location;
 
   const {InlineSelect} = Components
 
+  // permalinkedCommentHeight() finds the height of the comment at the top of comment permalink pages
+  // on the EA Forum and LessWrong. This is used when the user changes the comment sort order, which
+  // changes the page from a comment permalink page to a regular post page, and we want to preserve
+  // the user's (apparent) scroll position. With that top comment removed, we need to scroll up by its
+  // height.
+  // (If there is no permalinked comment at the top, whether because we're in a regular post page or
+  // because we're on Waking Up, which doesn't put permalinked comments at the top, this function
+  // returns zero.)
+  const permalinkedCommentHeight = () => {
+    const commentHeight = document.querySelector('.CommentPermalink-root')?.scrollHeight || 0;
+    const dividerEl = document.querySelector('.CommentPermalink-dividerMargins');
+    const dividerMarginBottom = dividerEl ? parseInt(getComputedStyle(dividerEl).marginBottom) : 0;
+    return commentHeight + dividerMarginBottom;
+  }
+
   const handleViewClick = (opt: Option & {value: CommentsViewName}) => {
     const view = opt.value
     const { query } = location;
     const currentQuery = isEmpty(query) ? {view: 'postCommentsTop'} : query
-    const newQuery = {...currentQuery, view: view, postId: post ? post._id : undefined}
+    const { commentId = undefined, ...newQuery } =  {...currentQuery, view: view, postId: post ? post._id : undefined}
```

Comment:
where do we use `commentId`? or is the point to remove it? if so I think this is a confusing way to do that and I'd do
`const newQuery = {...currentQuery, view: view, postId: post ? post._id : undefined, commentId: undefined}` instead

Reply:
yep that's better, I've changed it